cmd/medius/images/common.go

```diff
@@ -12,15 +12,22 @@ import (
 	"kubevirt.io/containerdisks/pkg/api"
 )
 
+const (
+	StagePush    = "push"
+	StageVerify  = "verify"
+	StagePromote = "promote"
+)
+
 type workerResult struct {
 	Key   string
 	Value api.ArtifactResult
 }
 
-func spawnWorkers(ctx context.Context, options *common.Options, workerFn func(api.Artifact) error) error {
+func spawnWorkers(ctx context.Context, options *common.Options, workerFn func(api.Artifact) (*api.ArtifactResult, error)) (chan workerResult, error) {
 	count := len(common.Registry)
 	errChan := make(chan error, count)
 	jobChan := make(chan api.Artifact, count)
+	resultsChan := make(chan workerResult, count)
```

Comment:
just add a `defer close(resultsChan)`, that is the preferrable pattern compared to the explicit close below.